bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/bridge/devices/HumiditySensorDevice.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.matter.internal.bridge.devices;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.matter.internal.bridge.MatterBridgeClient;
+import org.openhab.binding.matter.internal.client.dto.cluster.gen.RelativeHumidityMeasurementCluster;
+import org.openhab.core.items.GenericItem;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.MetadataRegistry;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.types.State;
+
+/**
+ * The {@link OnOffLightDevice}
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+public class HumiditySensorDevice extends GenericDevice {
+    private static final BigDecimal HUMIDITY_MULTIPLIER = new BigDecimal(100);
+
+    public HumiditySensorDevice(MetadataRegistry metadataRegistry, MatterBridgeClient client, GenericItem item) {
+        super(metadataRegistry, client, item);
+    }
+
+    @Override
+    public String deviceType() {
+        return "HumiditySensor";
+    }
+
+    @Override
+    public void handleMatterEvent(String clusterName, String attributeName, Object data) {
+    }
+
+    @Override
+    protected MatterDeviceOptions activate() {
+        primaryItem.addStateChangeListener(this);
+        MetaDataMapping primaryMetadata = metaDataMapping(primaryItem);
+        Map<String, Object> attributeMap = primaryMetadata.getAttributeOptions();
+        DecimalType state = primaryItem.getStateAs(DecimalType.class);
+        attributeMap.put(RelativeHumidityMeasurementCluster.CLUSTER_PREFIX + "."
+                + RelativeHumidityMeasurementCluster.ATTRIBUTE_MEASURED_VALUE, toMatterValue(state));
```

Comment:
Why getting a DecimalType while toMatterValue also supports QuantityType ?
I believe you could use `toMatterValue(primaryItem.getState())`, no ?